### PR TITLE
Combine movie details and overview fetching 

### PR DESCRIPTION
**Pull request by lihter**
I made api call in `MovieClient` in function `fetchMovieDetails(for movieId: Int) -> AnyPublisher<MovieDetailResponse, RequestError>`.
I still didn't make any changes to `NetworkClient` (Base client), but I plan to do so in the next PR, where I will also deal with adding FavoritesVC to the chain created in this PR.

For now, in `MovieDetailVC`, I've just written sinks viewDidLoad(), but later I will put all the sinks in the separate function called something like `fetchFromPresenter()`.

simulator_screenshot_DCB73B32-ABBD-4A20-862F-6F1E34C85724

### CODE REVIEW

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift [hunk 1]
```diff
@@ -61,7 +80,7 @@ extension MovieDetailViewController: MovieDetailDelegate {
     }
     
     func reloadData() {
-        titleView.reloadData()
+        //titleView.reloadData()
     }
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift:84]: no need for this anymore

File: MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClient.swift [hunk 2]
```diff
@@ -35,10 +36,23 @@ class MovieClient: MovieClientProtocol {
...
     }
     
-    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDetailResponse, RequestError>) -> Void) {
-        fetch(forUrl: "movie/\(movieId)") { (result: Result<MovieDetailResponse, RequestError>) in
-            completion(result)
+//    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDetailResponse, RequestError>) -> Void) {
+//        fetch(forUrl: "movie/\(movieId)") { (result: Result<MovieDetailResponse, RequestError>) in
+//            completion(result)
+//        }
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClient.swift:43]: no need for this anymore

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift [hunk 3]
```diff
@@ -33,6 +36,22 @@ class MovieDetailViewController: UIViewController {
         buildViews()
         
         presenter.fetchAll()
+
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift:39]: no need for new lines

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift [hunk 4]
```diff
@@ -24,39 +21,46 @@ final class MovieDetailPresenter {
...
+//
+//            switch result {
+//            case .success(let overview):
+//                self.delegate?.fillOverview(with: overview)
+//            case .failure(let error):
+//                print("Loading error: \(error.localizedDescription)")
+//            }
+//        }
+//    }
+
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift:63]: no need for this

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift [hunk 5]
```diff
@@ -24,39 +21,46 @@ final class MovieDetailPresenter {
...
+//            guard let self = self else { return }
+//
+//            switch result {
+//            case .success(let movie):
+//                self.delegate?.fillDetailTitleView(with: DetailTitleViewModel(fromModel: movie))
+//            case .failure(let error):
+//                print("Loading error: \(error.localizedDescription)")
+//            }
+//        }
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift:43]: no need for this

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift [hunk 6]
```diff
@@ -24,39 +21,46 @@ final class MovieDetailPresenter {
     }
     
     func fetchAll() {
-        getMovieDetails()
-        getOverview()
+        //getMovieDetails()
+        //getOverview()
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift:25]: delete

File: MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClientProtocol.swift [hunk 7]
```diff
@@ -10,7 +12,8 @@ protocol MovieClientProtocol {
     
     func fetchTopRatedTV(completion: @escaping(Result<[TVShowResponse], RequestError>) -> Void)
     
-    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDetailResponse, RequestError>) -> Void)
+    //func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDetailResponse, RequestError>) -> Void)
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClientProtocol.swift:15]: delete

File: MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift [hunk 8]
```diff
@@ -6,12 +7,19 @@ class UserDefaultsDataSource: UserDefaultsDataSourceProtocol {
...
-        } else {
-            return []
-        }
+//    var favorites: [Int] {
+//        if let favorites = UserDefaults.standard.object(forKey: favoritesUDKey) as? [Int] {
+//            return favorites
+//        } else {
+//            return []
+//        }
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift:16]: delete

File: MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift [hunk 9]
```diff
@@ -20,16 +28,16 @@ class UserDefaultsDataSource: UserDefaultsDataSourceProtocol {
...
         }
         UserDefaults.standard.setValue(favorites, forKey: favoritesUDKey)
     }
     
-    func isFavorite(movieId: Int) -> Bool {
-        favorites.contains(movieId)
-    }
+//    func isFavorite(movieId: Int) -> Bool {
+//        favorites.contains(movieId)
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift:41]: delete

File: MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSource.swift [hunk 10]
```diff
@@ -44,10 +46,17 @@ class MovieNetworkDataSource: MovieNetworkDataSourceProtocol {
...
     
-    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDataModel, RequestError>) -> Void) {
-        movieClient.fetchMovieDetails(for: movieId) { [weak self] result in
-            self?.mapMovieDetailResult(result: result, completion: completion)
-        }
+//    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDataModel, RequestError>) -> Void) {
+//        movieClient.fetchMovieDetails(for: movieId) { [weak self] result in
+//            self?.mapMovieDetailResult(result: result, completion: completion)
+//        }
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSource.swift:53]: delete

File: MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSourceProtocol.swift [hunk 11]
```diff
@@ -10,7 +12,8 @@ protocol MovieNetworkDataSourceProtocol {
     
     func fetchTopRatedTV(completion: @escaping(Result<[MovieDataModel], RequestError>) -> Void)
     
-    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDataModel, RequestError>) -> Void)
+    //func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieDataModel, RequestError>) -> Void)
```
Comment (fban1) [inline MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSourceProtocol.swift:15]: delete

File: MovieApp/MovieApp/App/Repository/MovieRepository.swift [hunk 12]
```diff
@@ -76,10 +77,25 @@ class MovieRepository: MovieRepositoryProtocol {
...
     
-    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieRepoModel, RequestError>) -> Void) {
-        networkDataSource.fetchMovieDetails(for: movieId) { [weak self] result in
-            self?.mapMovieDetailResult(result: result, completion: completion)
-        }
+//    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieRepoModel, RequestError>) -> Void) {
+//        networkDataSource.fetchMovieDetails(for: movieId) { [weak self] result in
+//            self?.mapMovieDetailResult(result: result, completion: completion)
+//        }
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/Repository/MovieRepository.swift:84]: delete

File: MovieApp/MovieApp/App/Repository/MovieRepository.swift [hunk 13]
```diff
@@ -164,38 +180,37 @@ class MovieRepository: MovieRepositoryProtocol {
...
+//
+//        userDefaultsDataSource.favorites.forEach { movieId in
+//            networkDataSource.fetchMovieDetails(for: movieId, completion: completionHandler)
+//        }
+        completion(.failure(.general))
     }
+//
+//    var favoriteMovies: AnyPublisher<[MovieRepoModel], Never> {
+//
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/Repository/MovieRepository.swift:213]: delete

File: MovieApp/MovieApp/App/Repository/MovieRepositoryProtocol.swift [hunk 14]
```diff
@@ -6,7 +8,8 @@ protocol MovieRepositoryProtocol {
     
     func fetchTopRatedMovies(completion: @escaping(Result<[MovieRepoModel], RequestError>) -> Void)
         
-    func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieRepoModel, RequestError>) -> Void)
+    //func fetchMovieDetails(for movieId: Int, completion: @escaping(Result<MovieRepoModel, RequestError>) -> Void)
```
Comment (fban1) [inline MovieApp/MovieApp/App/Repository/MovieRepositoryProtocol.swift:11]: delete

File: MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift [hunk 15]
```diff
@@ -25,22 +27,29 @@ class MoviesUseCase: MoviesUseCaseProtocol {
...
-            case .success(let movie):
-                completion(.success(movie.overview))
-            }
-        }
-    }
+//    func getMovieDetails(for movieId: Int, completion: @escaping(Result<MovieModel, RequestError>) -> Void) {
+//        moviesDataRepo.fetchMovieDetails(for: movieId) { [weak self] result in
+//            self?.mapMovieDetailResult(result: result, completion: completion)
+//        }
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift:35]: delete

File: MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift [hunk 16]
```diff
@@ -25,22 +27,29 @@ class MoviesUseCase: MoviesUseCaseProtocol {
...
+//    func getMovieOverview(for movieId: Int, completion: @escaping(Result<String, RequestError>) -> Void) {
+//        moviesDataRepo.fetchMovieDetails(for: movieId) { result in
+//            switch result {
+//            case .failure(let error):
+//                completion(.failure(error))
+//            case .success(let movie):
+//                completion(.success(movie.overview))
+//            }
+//        }
+//    }
```
Comment (fban1) [inline MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift:52]: delete

File: MovieApp/MovieApp/App/UseCase/MoviesUseCaseProtocol.swift [hunk 17]
```diff
@@ -6,9 +8,8 @@ protocol MoviesUseCaseProtocol {
         
     func getTopRatedMovies(completion: @escaping (Result<[MovieModel], RequestError>) -> Void)
         
-    func getMovieDetails(for movieId: Int, completion: @escaping(Result<MovieModel, RequestError>) -> Void)
-    
-    func getMovieOverview(for movieId: Int, completion: @escaping(Result<String, RequestError>) -> Void)
+    //func getMovieDetails(for movieId: Int, completion: @escaping(Result<MovieModel, RequestError>) -> Void)
```
Comment (fban1) [inline MovieApp/MovieApp/App/UseCase/MoviesUseCaseProtocol.swift:11]: delete

**Review by fban1 — CHANGES_REQUESTED** [review]

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift:84]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-f5b01144bf2cdd8175add69b29bdfdcf3abf149a38fdd4929f07da30b107f914L82-L84)

File: MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClient.swift (hunk 2 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClient.swift:43]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-0c98b99535b0a58a1773d03d140829a86847842a1642eeaf88fade3615fa98f5L39-L43)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift (hunk 3 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController.swift:39]: [Removed](https://github.com/lihter/Movie-App/pull/47/files#diff-f5b01144bf2cdd8175add69b29bdfdcf3abf149a38fdd4929f07da30b107f914L39)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift (hunk 4 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift:63]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-d8a7df924093b6e29d479b7644fb3593ce7762a26b79b7f10f290852dfb0ce54L51-L62)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift (hunk 5 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift:43]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-d8a7df924093b6e29d479b7644fb3593ce7762a26b79b7f10f290852dfb0ce54L32-L43)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift (hunk 6 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift:25]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-d8a7df924093b6e29d479b7644fb3593ce7762a26b79b7f10f290852dfb0ce54L24-L25)

File: MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClientProtocol.swift (hunk 7 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/NetworkClient/MovieClientProtocol.swift:15]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-0a228c21ed8c780e4987b127acc24fba8873d48e07d72d75abe28172ee51c8d5L15)

File: MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift (hunk 8 above)
Comment (lihter) [reply MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift:16]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-dfe654e503e62b61d6e962c2eec0634be5d54ff8715d6c83602c1572cd599349L10-L17)

File: MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift (hunk 9 above)
Comment (lihter) [reply MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift:41]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-dfe654e503e62b61d6e962c2eec0634be5d54ff8715d6c83602c1572cd599349L39-L41)

File: MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSource.swift (hunk 10 above)
Comment (lihter) [reply MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSource.swift:53]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-7d99464029c413709acb01d5e9eb9cc168b817201f2ba856f8b1212a3210a1c6L49-L53)

File: MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSourceProtocol.swift (hunk 11 above)
Comment (lihter) [reply MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSourceProtocol.swift:15]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-debe395bfd523b908607b9d04dc97f33cffbb590a2f04fc56df685dfa3d9b6d2L15)

File: MovieApp/MovieApp/App/Repository/MovieRepository.swift (hunk 12 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Repository/MovieRepository.swift:84]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-119fd471623eca33515c1b8fb62109b006cbeee1e1df1dc1a8e9a2e5ba5d4c27L80-L85)

File: MovieApp/MovieApp/App/Repository/MovieRepository.swift (hunk 13 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Repository/MovieRepository.swift:213]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-119fd471623eca33515c1b8fb62109b006cbeee1e1df1dc1a8e9a2e5ba5d4c27L210-L213)

File: MovieApp/MovieApp/App/Repository/MovieRepositoryProtocol.swift (hunk 14 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Repository/MovieRepositoryProtocol.swift:11]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-d71322365924cba479e79fb1ad727272478658da40f5becdf6183169df264316L11)

File: MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift (hunk 15 above)
Comment (lihter) [reply MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift:35]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-88fed2adfc92bb1f90d676d6f9e69ec5d3af18a8361903c9284e6d0e70098470L31-L35)

File: MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift (hunk 16 above)
Comment (lihter) [reply MovieApp/MovieApp/App/UseCase/MoviesUseCase.swift:52]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-88fed2adfc92bb1f90d676d6f9e69ec5d3af18a8361903c9284e6d0e70098470L43-L52)

File: MovieApp/MovieApp/App/UseCase/MoviesUseCaseProtocol.swift (hunk 17 above)
Comment (lihter) [reply MovieApp/MovieApp/App/UseCase/MoviesUseCaseProtocol.swift:11]: [Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-f372127c7d40320ce8dd547d5f9e2876cd89100a7824b0ce225657b790ff0fa3L11)

**Review by fban1 — APPROVED** [review]

**Pull request merged**